### DISCUSSION TITLE
[5.x] hide or set default state of certain variant fields

### DISCUSSION THREAD

**Discussion by bartrylant** [craftcms/commerce#3432] [Ideas] [answered]
For a lot if (if not all) product types on my site, I don't need all default variant fields and I would like to be able to hide them. E.g.: the weight field and the dimensions field.

It would also be nice to have a different default state for some field, for instance the inventory field. All my variants obviously have stock (also if it's 0, it counts as stock), so I would like the inventory field to be enabled by default.

Last but not least: a combination of those two. I want to have the "available for purchase" and "promotable" field to be enabled by default, but don't show the toggle button at all.

**Comment by wmdhosting** [comment]
Now there is field to hide  dimension. but does nto work..

**Comment by nfourtythree** [comment, marked as answer]
Hi @bartrylant 

You can hide fields the same way you would do for something like entries. For example, you could create a simple "Current User Condition" that would never get met and therefore it would be hidden for everyone.

See my example setup below:
![CleanShot 2024-05-01 at 09 57 50@2x](https://github.com/craftcms/commerce/assets/266453/b6e9592c-60bf-4dd8-9401-d5db69facf57)

Hope this helps, thanks!
